vcd/resource_vcd_vapp_vm.go

```diff
@@ -163,12 +251,31 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	var network *types.OrgVDCNetwork
+	var networks []*types.OrgVDCNetwork
+	var networksChanges []map[string]interface{}
 
 	if d.Get("network_name").(string) != "" {
-		network, err = addVdcNetwork(d, vdc, vapp, vcdClient)
+		network, err = addVdcNetwork(d.Get("network_name").(string), vdc, vapp, vcdClient)
 		if err != nil {
 			return err
 		}
+		if network != nil {
+			networks = append(networks, network)
+		}
+		networksChanges = append(networksChanges, map[string]interface{}{
+			"ip":         d.Get("ip").(string),
+			"orgnetwork": network.Name,
+		})
+	} else if d.Get("networks").([]interface{}) != nil {
```

Comment:
Can be simplified to a simple if because it is guarded by ConflictsWith?

```suggestion
	}
	if d.Get("networks").([]interface{}) != nil {
```